sdk/test/opentelemetry/sdk/trace/span_test.rb

```diff
@@ -200,14 +201,41 @@
       _(events.first.attributes).must_equal(attrs)
     end
 
-    it 'add event with timestamp' do
+    it 'uses the provided timestamp' do
       ts = Time.now
       span.add_event('added', timestamp: ts)
       events = span.events
       _(events.size).must_equal(1)
       _(events.first.timestamp).must_equal(exportable_timestamp(ts))
     end
 
+    it 'sets the end timestamp relative to the span start' do
+      timestamp = Time.new('2021-11-23 12:00:00.000000 -0600')
+      timestamp_in_nano = exportable_timestamp(timestamp)
+      clock_ids_expected = [Process::CLOCK_MONOTONIC, Process::CLOCK_REALTIME]
+      return_values = [500_000_000_000_000, timestamp_in_nano]
+      clock_gettime_mock = lambda do |clock_id, unit|
+        _(clock_id).must_equal(clock_ids_expected.shift)
+        _(unit).must_equal(:nanosecond)
+        return_values.shift
+      end
+
+      # Create a span with deterministic time values stored on it.
+      test_span = Process.stub(:clock_gettime, clock_gettime_mock) do
+        Span.new(context, Context.empty, OpenTelemetry::Trace::Span::INVALID, 'span', SpanKind::INTERNAL, nil, span_limits, [], nil, nil, timestamp, nil, nil)
+      end
+
+      Process.stub(:clock_gettime, 500_000_000_000_000 + 100) do
+        test_span.add_event('record something with relative time')
+        event = test_span.events[0]
+        _(event).wont_be_nil
+
+        # The expect timestamp is that of the parent offset by
+        # the drift in the monotonic clock
+        _(event.timestamp).must_equal(timestamp_in_nano + 100)
```

Comment:
✅ 2. Implicit Event timestamp is monotonic offset from realtime Span start timestamp